lib/matplotlib/tests/test_colors.py

```diff
@@ -156,6 +157,206 @@ def test_LogNorm():
+        norm = mcolors.MirrorPiecewiseNorm(fpos='crt', fneg='sqrt',
+                                           center_cm=0.35,
+                                           center_data=0.6)
+        assert_array_equal(norm([-2, 0.6, 1]), [0., 0.35, 1.0])
+
+    def test_fposfneg_refpoint_data(self):
+        norm = mcolors.MirrorPiecewiseNorm(fpos='crt', fneg='sqrt',
+                                           center_data=0.6)
+        assert_array_equal(norm([-2, 0.6, 1]), [0., 0.5, 1.0])
+
+    def test_fpos_lambdafunc(self):
+        norm = mcolors.MirrorPiecewiseNorm(fpos=lambda x: x**2,
+                                           fposinv=lambda x: x**0.5)
+        assert_array_equal(norm([-2, 0., 1]), [0., 0.5, 1.0])
+
+    def test_fpos_lambdafunc_fneg_string(self):
+        norm = mcolors.MirrorPiecewiseNorm(fpos=lambda x: x**2,
+                                           fposinv=lambda x: x**0.5,
+                                           fneg='crt')
+        assert_array_equal(norm([-2, 0., 1]), [0., 0.5, 1.0])
+
+    def test_intermediate_values(self):
+        norm = mcolors.MirrorPiecewiseNorm(fpos=lambda x: x**2,
+                                           fposinv=lambda x: x**0.5,
+                                           fneg='crt')
+        expected = [0.,
+                    0.1606978,
+                    0.52295918,
+                    0.68431122,
+                    1.]
+        assert_array_almost_equal(norm(np.linspace(-2, 3.5, 5)), expected)
+
+
+class TestMirrorRootNorm(object):
+    # All parameters except the order are just passed to the base class
+    def test_orderpos_only(self):
+        norm = mcolors.MirrorRootNorm(orderpos=2)
+        assert_array_equal(norm([-2, 0., 1]), [0., 0.5, 1.0])
+
+    def test_symmetric_default(self):
+        norm1 = mcolors.MirrorRootNorm(orderpos=3,
+                                       orderneg=3)
+        norm2 = mcolors.MirrorRootNorm(orderpos=3)
+        x = np.linspace(-2, 1, 10)
+        assert_array_equal(norm1(x), norm2(x))
+
+    def test_intermediate_values(self):
+        norm = mcolors.MirrorRootNorm(orderpos=3,
+                                      orderneg=4)
+        expected = [0.,
+                    0.0710536,
+                    0.23135752,
+                    0.79920424,
+                    0.89044833,
+                    0.95186372,
+                    1.]
+        assert_array_almost_equal(norm(np.linspace(-2, 3.5, 7)), expected)
+
+
+class TestRootNorm(object):
+    # All parameters except the order are just passed to the base class
+    def test_intermediate_values(self):
+        norm = mcolors.RootNorm(order=3)
+        expected = [0.,
+                    0.55032121,
+                    0.69336127,
+                    0.79370053,
+                    0.87358046,
+                    0.94103603,
+                    1.]
+        assert_array_almost_equal(norm(np.linspace(0, 10, 7)), expected)
+
+
 def test_PowerNorm():
```

Comment:
Haha, I do not blame you, it could perfectly be. In fact I have been tempted to re-implement it inheriting from PiecewiseNorm (It would not inherit from FuncNorm due to a difference in the [normalization policy](https://github.com/matplotlib/matplotlib/pull/7294#issuecomment-254897516) that @anntzer mentioned. (FuncNorm does the "LogNorm" way, because it is a rawer function, and PiecewiseNorm does it the "PowerNorm" way, because it really wants to ensure that the normalization function for each interval is bounded).

Again, my view on this would be to first merge the new classes, and give it a round of tests, and then maybe consider re-implementing other classes to make them inherit from these.